### ISSUE TITLE
Variables (locals) not shown when debugging MAUI app

### ISSUE THREAD

**Issue by ultra133** [microsoft/vscode-dotnettools#826]
When debugging MAUI apps on VS Code for Mac, and the app stops at a breakpoint, no locals are shown in the variables section. The section simply stays empty.

The call stack is displayed, and is is possible to add variables to the watch section. Also, the local variable's values can be shown by hovering over with the mouse.

It's the same behaviour on VS Code and VS Code Insiders and on all MAUI platforms (iOS, Android, ...).

When I debug a simple Net 8 console app, the local variables are displayed. So the issue is most likely related to the MAUI extension.

![image](https://github.com/microsoft/vscode-dotnettools/assets/3286258/83aaca08-c94f-4999-aed5-6d3a6f73c55f)

Extension version: 0.7.8
VS Code version: Code - Insiders 1.86.0-insider (Universal) (9621add46007f7a1ab37d1fce9bcdcecca62aeb0, 2023-12-20T05:38:07.020Z)
OS version: Darwin arm64 23.2.0
Modes:
| VS Bug <a href="https://devdiv.visualstudio.com/DevDiv/_queries/edit/1937480">1937480</a>

**Comment by durandt** [comment]
Thank you for reporting, I have just only realized I have the same issue and it makes debugging MAUI apps rather impossible. 

Also, when hovering or using watches, the string will be truncated after 100 characters... 

God this is long from being in parity with VS's (Windows or Mac) possibility to inspect variables and evaluate expressions.
Are we back to putting a bunch of Console.WriteLine all over the place?

**Comment by ultra133** [comment]
@durandt That's funny, Console.WriteLine doesn't work either: #685

**Comment by durandt** [comment]
@ultra133 On Android right? Because I was only testing on iOS today and it did show up (and I can see in #685 you referring to it being a problem on Android). Haven't tested that yet. Does it show up in `adb logcat` ? I think I've never really expected tooling wrapped around the android emulator to be able to capture the app's output specifically.

**Comment by T1m3rlan** [comment]
Same here. 
"Variables" are empty. 

osX, latest Visual Code release, latest stable MAUI.

**Comment by DaveMcCrady** [comment]
Same here. no locals for Android or IOS.
Version: 1.87.1 (Universal)
Commit: 1e790d77f81672c49be070e04474901747115651
Date: 2024-03-06T00:23:16.418Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.3.0
Maui Extension: v0.8.44
C# Dev Kit: v1.3.10

System.NotSupportedException: Specified method is not supported.
   at Mono.Debugging.Evaluation.NamespaceValueReference.get_Type() in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NamespaceValueReference.cs:line 62
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluatorVisitor.VisitInvocationExpression(InvocationExpressionSyntax node) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluatorVisitor.cs:line 833
   at Microsoft.CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor`1.Visit(SyntaxNode node)
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluatorVisitor.VisitInvocationExpression(InvocationExpressionSyntax node) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluatorVisitor.cs:line 802
   at Microsoft.CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor`1.Visit(SyntaxNode node)
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluator.Evaluate(EvaluationContext ctx, String expression, Object expectedType) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluator.cs:line 85
   at Mono.Debugging.Evaluation.ExpressionEvaluator.Evaluate(EvaluationContext ctx, String expression) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\ExpressionEvaluator.cs:line 43
   at Mono.Debugging.Evaluation.ObjectValueAdaptor.GetExpressionValue(EvaluationContext ctx, String exp) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\ObjectValueAdaptor.cs:line 1416
System.NotSupportedException: Specified method is not supported.
   at Mono.Debugging.Evaluation.NamespaceValueReference.get_Type() in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NamespaceValueReference.cs:line 62
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluatorVisitor.VisitInvocationExpression(InvocationExpressionSyntax node) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluatorVisitor.cs:line 833
   at Microsoft.CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor`1.Visit(SyntaxNode node)
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluatorVisitor.VisitInvocationExpression(InvocationExpressionSyntax node) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluatorVisitor.cs:line 802
   at Microsoft.CodeAnalysis.CSharp.Syntax.InvocationExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor`1.Visit(SyntaxNode node)
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluatorVisitor.VisitMemberAccessExpression(MemberAccessExpressionSyntax node) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluatorVisitor.cs:line 968
   at Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax.Accept[TResult](CSharpSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor`1.Visit(SyntaxNode node)
   at Mono.Debugging.Evaluation.NRefactoryExpressionEvaluator.Evaluate(EvaluationContext ctx, String expression, Object expectedType) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\NRefactoryExpressionEvaluator.cs:line 85
   at Mono.Debugging.Evaluation.ExpressionEvaluator.Evaluate(EvaluationContext ctx, String expression) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\ExpressionEvaluator.cs:line 43
   at Mono.Debugging.Evaluation.ObjectValueAdaptor.GetExpressionValue(EvaluationContext ctx, String exp) in D:\a\_work\1\s\external\debugger-libs\Mono.Debugging\Mono.Debugging.Evaluation\ObjectValueAdaptor.cs:line 1416

**Comment by DaveMcCrady** [comment]
This is becoming urgent.  It makes it next to impossible to use VS Code on mac to debug a maui app.  Any updates?

**Comment by ultra133** [comment]
No way, there has been no activity for 3 months now! This message means, the bot is about to close the ticket. Well done!
For those still looking for a solution: use Rider!

**Comment by mauroa** [comment]
I'm sorry you feel that way. In case you still want to use .NET MAUI in VS Code, I would like you to know that this issue should be already fixed if you update .NET MAUI to latest pre-release and also .NET SDK to at least 8.0.202.
We've introduced a new debug engine for MAUI apps (the same used by other C# projects), but it's available only from .NET 8.0.3 runtime (which is include from .NET SDK 8.0.202).

Please everyone confirm if you are unblocked after following the steps above.

Thanks

**Comment by durandt** [comment]
Hi @mauroa 
Good to hear this has been worked on. This was a big blocker to being able to develop & debug MAUI apps on VSCode.

> update .NET MAUI to latest pre-release

Do you mean the VSCode extension, or the MAUI NuGet packages (9.0.100-preview.3.10457)?

If the former, I should be able to try it tomorrow.

**Comment by mauroa** [comment]
@durandt I meant the .NET MAUI extension in VS Code. You might want to update the C# extension to latest pre-release as well, since some of the debugger stuff lives there.
Please let me know how that goes.
Thanks

**Comment by ultra133** [comment]
Thank you @mauroa! 

Well, like many others, I was finally forced to buy the Rider license for MAUI development, as VS Code still doesn't have a usable XAML editor, while VS for Mac became completely useless when it didn't support .Net 8, and MAUI developers were left with an unfinished VS Code IDE. But let's not stray off topic.
 
I can confirm that I'm finally seeing the local variables when debugging. Seems like you've fixed that!

**Comment by durandt** [comment]
Hi @mauroa,

I can confirm this is now fixed. Cool!

I tried first with pre-release VSCode extensions and with dotnet SDK 8.0.201 (forgot to update) and it did not work. Updating to latest SDK fixed it, as you wrote earlier. (I am aware this is no new information but always nice to confirm the requirement).

Now it would be sweet if [#880](https://github.com/microsoft/vscode-dotnettools/issues/880) could be fixed (although I have seen some improvements since the last time I tested that I will add to the issue)
